src/envoy/http/mixer/filter.cc

```diff
@@ -72,7 +72,9 @@ FilterHeadersStatus Filter::decodeHeaders(HeaderMap& headers, bool) {
   initiating_call_ = true;
   CheckData check_data(headers,
                        decoder_callbacks_->streamInfo().dynamicMetadata(),
-                       decoder_callbacks_->connection());
+                       decoder_callbacks_->connection(),
+                       decoder_callbacks_->streamInfo().routeEntry()->metadata(),
+                       decoder_callbacks_->clusterInfo()->metadata());
```

Comment:
Can a route not have an upstream, e.g. direct response action? If so, this needs to be null-protected.

Reply:
Moved to report data and added some checks. PTAL.